examples/get_matching/server.c

```diff
@@ -0,0 +1,188 @@
+		*ret_value = 1;
+	}
+	ret = PtlEQWait(eqh, &ev);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		*ret_value = 1;
+	}
+	if (ret_value == 0) {
+		PtlEvToStr(0, &ev, msg);
+		printf("Event : %s \n", msg);
+	}
+}
+
+int main(int argc, char **argv)
+{
+	int res;
+	int ret;
+	ptl_handle_ni_t nih;
+	ptl_index_t pti;
+	ptl_handle_eq_t eqh;
+	ptl_process_t id;
+	ptl_event_t ev;
+	ptl_handle_me_t meh;
+	int ret_value = 0;
+	int n = 1;
+	char msg[PTL_EV_STR_SIZE];
+	ptl_process_t id_server = { .phys = { .pid = SERVER_PID, .nid = 1 } };
+	char data[MESSAGELENGTH * 10];
+	ptl_me_t me;
+	bool wait_for_ever = false;
+	long num_messages = 0;
+	char *p;
+
+	if (argc < 2) {
+		wait_for_ever = true;
+	} else {
+		num_messages = strtol(argv[1], &p, 10);
+		if (*p != '\0') {
+			fprintf(stderr,
+				"An invalid character was found before the end of the string");
+			return 1;
+		}
+	}
+
+	for (int i = 0; i < 10; i++)
+		sprintf(&data[i * MESSAGELENGTH], "message%d !", i);
+
+	printf(" \n--- Server interface --- \n \n");
+
+	ret = PtlInit();
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlInit failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	}
+
+	ret = PtlNIInit(PTL_IFACE_DEFAULT, PTL_NI_NO_MATCHING | PTL_NI_PHYSICAL, SERVER_PID, NULL,
+			NULL, &nih);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlNIInit failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto fini;
+	}
+
+	ret = PtlEQAlloc(nih, 10, &eqh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto ni_fini;
+	}
+
+	ret = PtlPTAlloc(nih, 0, eqh, 0, &pti);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlPTAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto eq_free;
+	}
+
+	/* get server pid and server nid so the client know the server id */
+	ret = PtlGetId(nih, &id);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlGetId failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto pt_free;
+	}
+	printf("Server Pid : %d \n", id.phys.pid);
+	printf("Server Nid : %d \n\n", id.phys.nid);
+
+	/* describe the memory used to send messages */
+	me = (ptl_me_t){ .start = data,
+			 .length = sizeof(data),
+			 .ct_handle = PTL_CT_NONE,
+			 .uid = PTL_UID_ANY,
+			 .options = PTL_LE_OP_GET | PTL_ME_MANAGE_LOCAL,
+			 .match_id = id_server,
+			 .match_bits = 0,
+			 .ignore_bits = 0,
+			 .min_free = MESSAGELENGTH - 1 };
+
+	add_me(nih, pti, me, &meh, eqh, ev, msg, &ret_value);
+	if (ret_value == 1) {
+		res = 1;
+		goto me_unlink;
+	}
+
+	while (num_messages > 0 || wait_for_ever) {
+		ret = PtlEQWait(eqh, &ev);
+		if (ret != PTL_OK) {
+			fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+			res = 1;
+			goto me_unlink;
+		}
+
+		PtlEvToStr(0, &ev, msg);
+		printf("Event : %s \n", msg);
+
+		if (ev.type == PTL_EVENT_AUTO_UNLINK) {
+			n = (n + 1) % 10;
+			add_me(nih, pti, me, &meh, eqh, ev, msg, &ret_value);

```

Comment:
The me should already be unlinked  as we received the PTL_EVENT_AUTO_UNLINK, we should goto pt_free in the failure case I think